In [1]:
import numpy as np

## Se tiene la siguiente muestra X = {4, 3, 11, 5, 4, 50, 6, 8, 7, 9}, en el que el valor 50 es un outlier.

### a) Calcula la media recortada de la muestra al 20 %, al 10 % y calcula la centrimedia.

Para calcular las medias recortadas, se eliminan los datos una vez ordenados por
cada extremo y se calcula la media. para recortar un 20 % (α = 0.2), se recorta
un 10 % (un valor) en cada extremo y se calcula la media. Para recortar un 10 %
(α = 0.1), se pondera los valores de los extremos con 0.5 y se calcula la media
considerando que n es igual a 9. Por último, para obtener la centrimedia se eliminan
2 valores de cada extremo, se pondera por 0.5 el tercer y el octavo valor y se obtiene
la media considerando n = 5.

In [2]:
#se define la muestra
X = np.array([4, 3, 11, 5, 4, 50, 6, 8, 7, 9])

#ordenar los valores de la muestra
X.sort()

X

array([ 3,  4,  4,  5,  6,  7,  8,  9, 11, 50])

In [3]:
# quitar valores recortados 0.1----------------------------------
X1 = X[1:9]

X1

array([ 4,  4,  5,  6,  7,  8,  9, 11])

In [4]:
print("La media recortada al 20 % es", sum(X1)/8)

La media recortada al 20 % es 6.75


In [5]:
X2 = np.array([3*0.5,4,4,5,6,7,8,9, 11, 50*0.5])
print("La media recortada al 10 % es",round(sum(X2)/9, 2))

La media recortada al 10 % es 8.94


In [6]:
X3 = X[2:8]
print(X3)
X3[0] = 4*0.5
X3[5] = 8*0.5
print(X3)
print("La centrimedia es", sum(X3)/5)

[4 5 6 7 8 9]
[2 5 6 7 8 4]
La centrimedia es 6.4


### b) Calcula la media winsorizada al 20 %.

Para obtener la media winsorizada, en lugar de eliminar las observaciones, se susti-
tuyen por los últimos valores. Es análogo a las medias recortadas excepto en que las
puntuaciones que se eliminaban ahora se sustituyen por los valores menor y mayor
que quedan para el cómputo de la media winsorizada.

In [7]:
# definir la muestra---------------------------------------------
X = np.array([4, 3, 11, 5, 4, 50, 6, 8, 7, 9])
# ordenar los valores de la muestra------------------------------
X.sort()
X

array([ 3,  4,  4,  5,  6,  7,  8,  9, 11, 50])

In [8]:
# quitar valores recortados 0.1----------------------------------
W1 = np.array([4,4,4,5,6,7,8,9,11,11])
# print(W1)
print("La media winsorizada es", sum(W1)/10)

La media winsorizada es 6.9


### c) Calcula la trimedia.

La trimedia se obtiene ponderando la mediana y los primer y tercer cuartiles según la fórmula:

trimedia = (cuartil1 + 2 · mediana + cuartil3)/4

In [9]:
X = np.array([4, 3, 11, 5, 4, 50, 6, 8, 7, 9])
X.sort()
X

array([ 3,  4,  4,  5,  6,  7,  8,  9, 11, 50])

In [10]:
q1_x = np.quantile(X, 0.25, interpolation='midpoint')
me_x = np.quantile(X, 0.50, interpolation='midpoint')
q3_x = np.quantile(X, 0.75, interpolation='midpoint')

trimedia = (q1_x + 2 * me_x + q3_x)/4
trimedia

6.5

### d) Compara los valores obtenidos en los apartados a), b) y c) con la media muestral.

In [11]:
# definir la muestra---------------------------------------------
X = np.array([4, 3, 11, 5, 4, 50, 6, 8, 7, 9])
# ordenar los valores de la muestra------------------------------
print("La media muestral es", X.mean())

La media muestral es 10.7


La media muestral es 10.7, que es mucho más sensible al outlier de 50, que los métodos
robustos obtenidos en los apartados a, b y c.

### e) Calcula el MAD y comparalo con la desviación típica.

Para calcular el MAD es necesario calcular la mediana de la diferencia absoluta de
cada valor con la mediana de la muestra.

In [12]:
print("La mediana de la muestra vale me_x = ", me_x)

La mediana de la muestra vale me_x =  6.5


In [13]:
## La mediana de la muestra vale me_x =
6.5
AD = abs(X - me_x)
MAD = np.quantile(AD, 0.5, interpolation = "midpoint")
print("El MAD vale", MAD)

El MAD vale 2.5


In [35]:
print("La desviación típica es", round(X.std(), 1))

La desviación típica es 13.3


Se observa que el MAD es mucho más insensible a valores atípicos que la desviación
típica de la muestra.